core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -267,7 +269,9 @@ export class TopicEditorNavbarComponent {
     this.warningsAreShown = false;
     this.showTopicEditOptions = false;
     this.topic = this.topicEditorStateService.getTopic();
-    this.topicSkillIds = this.topic.getSkillIds();
+    if (this.topic) {
```

Comment:
Suppose there is no topic at the time of init, and the topic loads later. Do we call ngOnInit() again (or some other method that initializes the topic-related stuff properly)?

Reply:
Now I have added the code get skill ids whenever topic is initialized

Reply:
But my question still stands. Let me try to break it down: the main question is, is it conceptually possible for there to be no topic at time of init?

If so, we need to initialize this.topic and this.topicSkillIds to null and [] respectively, and add a comment above them saying when exactly they'll be populated.

If not then we should throw an error if getTopic() returns null at this point.

Reply:
Yes, you should alter the tests in the spec file accordingly -- see comment above.

Reply:
Done